### PR TITLE
chore: add lint on gh actions

### PR DESCRIPTION
**Pull request by fmoral2**
#### Proposed Changes ####

- add go lint on pushes 

#### Types of Changes ####

- Addition of golang ci lint on pushes 

#### Verification ####

- Pushing new code

#### Testing ####
- n/a

### CODE REVIEW

File: .github/workflows/run-distros.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+on:
+  push:
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: Build and Lint
+    runs-on: ubuntu-20.04
```
Comment (VestigeJ) [inline .github/workflows/run-distros.yaml:12]: Do we want to keep these ubuntu-latest for the freemium run time?

Comment (fmoral2) [reply .github/workflows/run-distros.yaml:12]: Any objections?

Comment (fmoral2) [reply .github/workflows/run-distros.yaml:12]: I just got the default one from actions, I can add other

File: scripts/Dockerfile.build [hunk 2]
```diff
@@ -22,6 +22,8 @@ RUN apk update && \
     chmod +x /usr/local/bin/terraform && \
     rm -rf tmp
 
+RUN curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /usr/local/bin v1.53.3
```
Comment (osodracnai) [inline scripts/Dockerfile.build:25]:
```suggestion
```
actions will already run golangci-liint, install on docker is not needed

Comment (fmoral2) [reply scripts/Dockerfile.build:25]: added there to have available running here locally with docker

File: .github/workflows/run-distros.yaml (hunk 1 above)
Comment (VestigeJ) [reply .github/workflows/run-distros.yaml:12]: doing ubuntu-latest keeps it a free run if we’re under our monthly allowance from github for the org if i recall correctly.

Comment (fmoral2) [reply .github/workflows/run-distros.yaml]: sure

**Comment by mdrahman-suse** [comment]
I don't see any step for cleanup, is that not needed or will be addressed later?

**Review by VestigeJ — APPROVED** [review]
LGTM

File: scripts/Dockerfile.build (hunk 2 above)
Comment (mdrahman-suse) [reply scripts/Dockerfile.build:25]: In that case does it makes sense to have separate dockerfile for local and CI?

**Review by mdrahman-suse — APPROVED** [review]
LGTM

**Pull request merged**